src/main/java/io/antmedia/streamsource/StreamFetcherManager.java

```diff
@@ -89,7 +90,12 @@ public void setRestartStreamFetcherPeriod(int restartStreamFetcherPeriod) {
 
 
 	public StreamFetcher startStreaming(Broadcast broadcast) {	
-
+		if(scope.getContext().hasBean(IResourceMonitor.BEAN_NAME)) {
+			IResourceMonitor monitor = (IResourceMonitor) scope.getContext().getBean(IResourceMonitor.BEAN_NAME);
+			if(monitor.getAvgCpuUsage() > monitor.getCpuLimit()) {
+				return null;
+			}
+		}
```

Comment:
startStreaming is called from rest service as well so that we need to return meaningful errors in order to have better error handling

Reply:
You are right but we need some refactoring to handle the situation. I recommend creating an issue for such error handling.